### DISCUSSION TITLE
Question about Laziness with Field Comprehensions

### DISCUSSION THREAD

**Discussion by PierreR** [cue-lang/cue#1133] [Q&A]
Given a list of struct defined by:

deployment: [ID=_] {
}

There seems to be a subtle difference (laziness ?) between:
```
for d in [deployment] for _, v in d {
}
```
And the more simple form:
```
for _,v in deployment {
}
```

I have recently realized that both forms are not equivalent. The simple one was failing exposing a bug in my configuration while the longer form was working keeping a bug from showing up.

Thanks for your help

**Comment by myitcv** [comment]
@PierreR did you happen to spot this as a result of looking at https://github.com/cue-lang/cue/discussions/1134?

**Reply by PierreR** [reply]
@myitcv  yes. So probably a side effect of it.

So I guess it makes sense to always use the simpler `for _,v in deployment {` and I can mark this discussion as answered ?

**Reply by myitcv** [reply]
My sense is that there is a bug lurking here somewhere. Do you have a repro?
